xline/src/storage/kvwatcher.rs

```diff
@@ -332,3 +342,84 @@ impl WatchEvent {
         std::mem::take(&mut self.events)
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    use std::{collections::BTreeMap, time::Duration};
+
+    use tokio::time::timeout;
+    use utils::config::StorageConfig;
+
+    use crate::{
+        header_gen::HeaderGenerator,
+        rpc::{PutRequest, RequestWithToken},
+        storage::{db::DB, index::Index, lease_store::LeaseCollection, KvStore},
+    };
+
+    use super::*;
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+    async fn it_works() {
+        let (store, db, kv_watcher) = init_empty_store();
+        let mut map = BTreeMap::new();
+        let handle = tokio::spawn({
+            let store = Arc::clone(&store);
+            async move {
+                for i in 0..100_u8 {
+                    let req = RequestWithToken::new(
+                        PutRequest {
+                            key: "foo".into(),
+                            value: vec![i],
+                            ..Default::default()
+                        }
+                        .into(),
+                    );
+                    let (sync_res, ops) = store.after_sync(&req).await.unwrap();
+                    db.flush_ops(ops).unwrap();
+                    store.mark_index_available(sync_res.revision());
+                }
+            }
+        });
+        tokio::time::sleep(std::time::Duration::from_micros(500)).await;
+        let (event_tx, mut event_rx) = mpsc::channel(128);
+        kv_watcher.watch(123, KeyRange::new_one_key("foo"), 1, vec![], event_tx);
+
+        'outer: while let Some(event_batch) = timeout(Duration::from_secs(3), event_rx.recv())
+            .await
+            .unwrap()
+        {
+            for event in event_batch.events {
+                let val = event.kv.as_ref().unwrap().value[0];
+                let e = map.entry(val).or_insert(0);
+                *e += 1;
+                if val == 99 {
+                    break 'outer;
+                }
+            }
+        }
+
+        assert_eq!(map.len(), 100);
+        for (k, count) in map {
+            assert_eq!(count, 1, "key {k} should be notified once");
+        }
+        handle.abort();
+    }
+
+    fn init_empty_store() -> (Arc<KvStore<DB>>, Arc<DB>, Arc<KvWatcher<DB>>) {
```

Comment:
Shall we move the util function to the top of test mod?

Reply:
Moved